rfcs/0057-named-ellipses.md

```diff
@@ -0,0 +1,36 @@
+---
+feature: Named Ellipses OR function argument consistency
+start-date: 2019-11-14
+author: deliciouslytyped
+co-authors: none
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues:
+  Named Ellipses - https://github.com/NixOS/nix/issues/2998
+---
+
+# Summary
+[summary]: #summary
+
+It should be possible to bind a name to ellipses in a function definition like `{ a, ...@extra }: null`, and `{ a, extra@... }: null`. This makes intuitive sense, and could remove the need for a lot of uses of `removeAttrs` that really just want to refer to the contents of ellipses.
+
+TODO: The latter could be substantiated.
+
+# Detailed design
+[design]: #detailed-design
+
+`{...@extraargs}: extraargs` should yield as an attrset the extra arguments "in" `...`.
+
+TODO: consider what other languages like Haskell do
```

Comment:
Note that you want to consider pattern matching libraries, too, not just function arguments.

Some languages use notations that wouldn't fit well with the current use of `...` in Nix: `def f(a,b,*args):` in Python, `(lambda (a b &rest args) …)` in Common Lisp.